src/components/elements/tableFilters/filters/ColumnsMenu.tsx

```diff
@@ -0,0 +1,132 @@
+import ViewColumnIcon from '@mui/icons-material/ViewColumn';
+import {
+  Box,
+  Button,
+  Checkbox,
+  Divider,
+  FormControlLabel,
+  FormGroup,
+  Popover,
+  Stack,
+  Typography,
+} from '@mui/material';
+import { isEmpty } from 'lodash-es';
+import {
+  usePopupState,
+  bindTrigger,
+  bindMenu,
+} from 'material-ui-popup-state/hooks';
+import { ReactNode, useState } from 'react';
+
+import TableFilterButton from './FilterButton';
+
+export interface TableColumnsMenuProps {
+  columns: { value: string; header: ReactNode; defaultHidden: boolean }[];
+  columnsValue: string[];
+  setColumnsValue: (value: string[]) => any;
+}
+
+const TableColumnsMenu = ({
+  columns,
+  columnsValue,
+  setColumnsValue,
+}: TableColumnsMenuProps): JSX.Element => {
+  const [intermediateValue, setIntermediateValue] =
+    useState<typeof columnsValue>(columnsValue);
+  const popupState = usePopupState({
+    variant: 'popover',
+    popupId: 'sortMenu',
+  });
+
+  return (
+    <>
+      <TableFilterButton
+        startIcon={<ViewColumnIcon />}
+        active={!isEmpty(columnsValue)}
+        {...bindTrigger(popupState)}
+      >
+        Columns
+      </TableFilterButton>
+      <Popover
+        {...bindMenu(popupState)}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+      >
+        <Box p={2}>
+          <Typography variant='overline'>Show Columns</Typography>
+          <FormGroup>
+            {columns.map(({ value, header }) => (
+              <FormControlLabel
+                key={value}
+                control={
+                  <Checkbox checked={intermediateValue.includes(value)} />
+                }
+                onChange={(e, checked) =>
+                  setIntermediateValue(
+                    checked
+                      ? [...intermediateValue, value]
+                      : intermediateValue.filter((v) => v !== value)
+                  )
+                }
+                label={header}
+              />
+            ))}
+          </FormGroup>
+          <Divider sx={{ mt: 3 }} />
+          <Stack direction='row' gap={1} p={2}>
```

Comment:
It looks like there's a lot of repeated code here between the ColumnsMenu and the FilterMenu/FilterContent. 
It could use a refactor so that they both use the same Popover. Similar to how they share `TableFilterButton`, can they share a `TableControlPopover` that encapsulates the design (popover, padding, title, divider, action buttons), and accepts action handlers and the content as props?

Reply:
(And lets keep the styling of the Filter menu for both. For example the Filter menu has less padding below the buttons than the Columns menu does.)